lib/rules/next-tick-style.js

```diff
@@ -123,6 +159,61 @@ module.exports = {
           return
         }
 
+        if (preferredStyle === 'await') {
+          if (
+            callExpression.arguments.length > 0 ||
+            !isAwaitedFunction(callExpression)
+          ) {
+            context.report({
+              node,
+              messageId: 'useAwait',
+              fix(fixer) {
+                const sourceCode = context.getSourceCode()
+
+                // Handle callback to await conversion
+                if (callExpression.arguments.length > 0) {
+                  const [args] = callExpression.arguments
+                  let callbackBody = null
+
+                  callbackBody =
+                    args.type === 'ArrowFunctionExpression' ||
+                    args.type === 'FunctionExpression'
+                      ? extractCallbackBody(args, sourceCode)
+                      : `${sourceCode.getText(args)}()`
+
+                  const nextTickCaller = sourceCode.getText(
+                    callExpression.callee
+                  )
+                  return fixer.replaceText(
+                    callExpression.parent,
+                    `await ${nextTickCaller}();${callbackBody};`
```

Comment:
We need make sure the `nextTick` call is an expression statement, otherwise this will cause sytax errors.


```
if (foo) nextTick(...)
```

```
(() => nextTick(...))();
```

Reply:
@fisker How would you suggest the fix method handle these cases?

Reply:
Ignore or turn into a block.

Reply:
"ignore" means no fix

Reply:
Okay, I've added a check.

Reply:
Please review the PR when you get the chance and let me know if there's any other changes you think I should make. Thanks

Reply:
https://github.com/vuejs/eslint-plugin-vue/pull/2508#discussion_r1677865772

Reply:
Isn't it related to this one: https://github.com/vuejs/eslint-plugin-vue/pull/2508#discussion_r1677868269 ?
I removed the `.slice()` method which keeps the block

Reply:
Sorry didn't see that change.

Reply:
No worries. Is everything good to go now?

Reply:
I'm not a maintainer..

Reply:
This seems not the correct fix.. It changes behavior

```js
nextTick(() => console.log(2))
console.log(1)
```

Logs `1 2`

```js
await nextTick();  console.log(2)
console.log(1)
```

Logs `2 1`

Maybe use suggestions instead?